docs/week-3.md

```diff
@@ -0,0 +1,523 @@
+
+# Jumpstart Week 3: React and Material-UI
+
+Welcome to the 3rd week of the Jumpstart Program! This is the week we get into the good stuff...React! 
+This word is probably one that you have heard whenever Web Development has been the topic of conversation 
+and now you will get to learn about this awesome tool.
+
+## React
+Our good friend John wants to make an interactive user interface for his website but does not want to go through
+the whole process of Creating an HTML layout and then applying a CSS stylesheet and finally adding in Javascript
+functionality and liveliness. If only there was a simple library that can do all of this for us and more.......oh
+wait there is. It is called React and it is dope.
+ 
+### What is React?
+React is a declarative, efficient, and flexible JavaScript library for building user interfaces. 
+It lets you compose complex User Interfaces from small and isolated pieces of code called “components”. 
+React basically will build you a website out of pure Javascript! It combines HTML tags in the form of JSX, 
+CSS files and good old JS logic to produce a masterpiece. 
+
+
+### Why is React better than plain old HTML/CSS/JS combo?
+Well React is just a Javascript Library which requires less knowledge and understanding of other frameworks. 
+In addition, React makes it easy for developers to manipulate and work with the DOM with the help of a virtual browser. 
+The virtual browser bridges the gap between the developer and the browser (the actual one). Lastly, React uses a type
+of code called **declarative code**. Declarative code describes the end result, but does not act as a step-by-step 
+guide of how to do it. In practice, that means declarative code is lightweight, easier to understand and change, 
+and has fewer bugs. However, declarative code is not so good preventing errors which is why we use Typescript to help
+prevent errors that may break our application.
+
+### What a file looks like that incorporates React
+Let us call this file LuckyNum.tsx (.tsx refers to the file being a typescript file).
+```
```

Comment:
Also add a space between code blocks and anything else (MkDocs/GitHub can be a little weird about that sometimes)